### PR TITLE
Refactors the detective scanner to fix potentially getting stuck in the 'scanning' state

### LINKED ISSUE #18241: Detective scanner gets "stuck" trying to scan something and bricks
Reporting client version: 514.1589

## Round ID:
7250

## Reproduction:
Use the detective scanner a handful of times. it works for a bit, then gets stuck and never works again. If you try to print reports it just says "scanner busy"

### PR DESCRIPTION
**Pull request by vinylspiders**
## About The Pull Request

Detective scanners make use of a `scanning` boolean to determine whether or not you can perform another action with it. 
It gets set to `FALSE` in its procs, but if any of those procs runtime for whatever reason this could lead to that never happening.

Which in turn leads to it getting stuck in the `scanning = TRUE` state indefinitely.

---

This PR just moves the functions into wrapper functions so that should a runtime occur, the boolean will still get set to `FALSE`. Also changed the boolean's name from `scanning` to `scanner_busy` because it is also used for the print function.

Fixes https://github.com/Skyrat-SS13/Skyrat-tg/issues/18241

## Why It's Good For The Game

Code resiliency

## Changelog

:cl:
fix: fixes a potential bug that can cause the detective scanner to get stuck in a scanning state indefinitely
/:cl:

### CODE REVIEW

**Review by Mothblocks — CHANGES_REQUESTED** [review]
dont think we need a balloon alert on this at all especially since we already report runtimes

**Review by san7890 — CHANGES_REQUESTED** [review]
Whoops, forgot to hit send even though I reviewed this earlier today. You just did a lot of code moving, and I figured that we can probably do some housekeeping/code improvment while in the area (especially since you're billing this as a type of refactor in the title of your PR).

Feel free to mark this PR ready for review when these have been addressed in some manner.

File: code/modules/detectivework/scanner.dm [hunk 1]
```diff
@@ -63,124 +73,141 @@
...
+	det_data[DETSCAN_CATEGORY_FIBER] = GET_ATOM_FIBRES(scanned_atom)
+
+	var/target_name = scanned_atom.name
+
+	// Start gathering
+
+	if(ishuman(scanned_atom))
+
+		var/mob/living/carbon/human/H = scanned_atom
+		if(!H.gloves)
```
Comment (san7890) [inline code/modules/detectivework/scanner.dm:134]: Clean up this single-letter var.

**Review by san7890 — APPROVED** [review]
Looks good to me after this is addressed. Sorry for not spotting this sooner.

**Pull request merged**